crates/core/src/tls.rs

```diff
@@ -109,3 +110,30 @@ impl NativeRootsExt for reqwest::ClientBuilder {
             .fold(self, reqwest::ClientBuilder::add_root_certificate)
     }
 }
+
+pub fn load_certs_from_paths(
+    paths: &[std::path::PathBuf],
+) -> anyhow::Result<Vec<rustls::pki_types::CertificateDer<'static>>> {
+    paths
+        .iter()
+        .map(read_certs_from_path)
+        .flat_map(|result| match result {
+            Ok(vec) => vec.into_iter().map(Ok).collect(),
+            Err(er) => vec![Err(er)],
+        })
+        .collect::<Result<Vec<_>, _>>()
+}
+
+pub fn read_certs_from_path(
+    path: &std::path::PathBuf,
+) -> anyhow::Result<Vec<rustls::pki_types::CertificateDer<'static>>> {
+    // TODO(joonas): Support directories
+    if path.is_file() {
+        let f = std::fs::File::open(path)
+            .with_context(|| format!("failed to open file at provided path: {}", path.display()))?;
+        let mut reader = std::io::BufReader::new(f);
+        Ok(rustls_pemfile::certs(&mut reader).collect::<Result<Vec<_>, _>>()?)
+    } else {
+        Ok(vec![])
+    }
```

Comment:
```suggestion
    let path = path.as_ref();
    // TODO(joonas): Support directories
    if !path.is_file() {
        return Ok(Vec::with_capacity(0));
    }
    let f = std::fs::File::open(&path)
        .with_context(|| format!("failed to open file at provided path: {}", path.display()))?;
    let mut reader = std::io::BufReader::new(f);
    Ok(rustls_pemfile::certs(&mut reader).collect::<Result<Vec<_>, _>>()?)
```

*if* you want to pull in that `AsRef<Path>` change... 

Take it or leave it :)